x-pack/plugins/enterprise_search/server/lib/ml/fetch_ml_models.ts

```diff
@@ -80,11 +80,10 @@ export const fetchMlModels = async (
 
   // Undeployed placeholder models might be in the Downloading phase; let's evaluate this with a call
   // We must do this one by one because the API doesn't support fetching multiple models with include=definition_status
-  for (const model of models) {
-    if (model.isPromoted && !model.isPlaceholder && !model.hasStats) {
-      await enrichModelWithDownloadStatus(model, trainedModelsProvider);
-    }
-  }
+  const enrichmentCalls = models
+    .filter((model) => model.isPromoted && !model.isPlaceholder && !model.hasStats)
+    .map((model) => enrichModelWithDownloadStatus(model, trainedModelsProvider));
+  await Promise.all(enrichmentCalls);
```

Comment:
question: Can we do something better than throwing out of the function if one of these calls errors out and the others succeed?

Reply:
I'm happy to add a `.catch()` to the promise chain inside `enrichModelWithDownloadStatus()` and handle the rejection case with a no-op. This means a failed call will set the wrong status for its model ("not deployed" instead of "dowloading") but I think that's fine.

Reply:
@sphilipse I added graceful handling for when a model status fetch is failing: https://github.com/elastic/kibana/pull/177375/commits/81d0e808643a77e713eeab3db92da8e1bfb7a51b